packages/core/src/strategies/create2-strategy.ts

```diff
@@ -82,14 +132,22 @@ export class Create2ExecutionStrategy implements ExecutionStrategy {
     }
 
     // On a local hardhat node, deploy the CreateX factory
-    await this._deployCreateXFactory(this.client);
+    await this._deployCreateXFactory(this._jsonRpcClient);
   }
 
   public async *executeDeployment(
     executionState: DeploymentExecutionState
   ): DeploymentStrategyGenerator {
-    const artifact = await this._loadArtifact(executionState.artifactId);
-    const salt = ethers.id("test");
+    assertIgnitionInvariant(
+      this._deploymentLoader !== undefined && this._jsonRpcClient !== undefined,
+      "Strategy not initialized"
+    );
+
+    const artifact = await this._deploymentLoader.loadArtifact(
+      executionState.artifactId
+    );
+
+    const salt = ethers.id(this.config.salt);
```

Comment:
Is this worth wrapping in a util function to encapsulate the `ethers` usage?

Reply:
Why are we hashing this? Doesn't createX do it? is this expected? Can't we validate that it's 32bytes instead? Doesn't this alter the input, making it impossible to control the replay protection with bit flags as createX expects?

I know the answers to some of this questions are "no", because you have some tests using those bit flags, but i don't get it.